repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -83,4 +84,32 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
     return session.state.toString
   }
 
+  override def createWrapper(msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
+    kind match {
+      case Spark() => super.createWrapper(msg)
+      case PySpark() | PySpark3() => {
+        new BypassJobWrapper(this, msg.id,
+          new BypassPySparkJob(msg.serializedJob, PythonInterpreter.getPySparkJobProcessor))
+      }
+    }
+  }
+
+  override def addFile(path: String): Unit = {
+    kind match {
+      case Spark() => super.addFile(path)
+      case PySpark() | PySpark3() => PythonInterpreter.getPySparkJobProcessor.addFile(path)
+    }
+  }
+
+  override def addJarOrPyFile(path: String): Unit = {
+    kind match {
+      case Spark() => super.addJarOrPyFile(path)
+      case PySpark() | PySpark3() => {
+        val localCopyDir = new File(PythonInterpreter.getPySparkJobProcessor.getLocalTmpDirPath)
```

Comment:
In fact you don't even need to expose it at all. Can't you just add the path returned by `doAddJarOrPyFile_copyFileToLocal` instead?